mso/resource_mso_remote_location.go

```diff
@@ -0,0 +1,247 @@
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSORemoteLocation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSORemoteLocationCreate,
+		Update: resourceMSORemoteLocationUpdate,
+		Read:   resourceMSORemoteLocationRead,
+		Delete: resourceMSORemoteLocationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceMSORemoteLocationImport,
+		},
+
+		SchemaVersion: version,
+
+		Schema: (map[string]*schema.Schema{
+
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"scp",
+					"sftp",
+				}, false),
+			},
+			"hostname": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"path": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"port": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  22,
+			},
+			"username": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"password": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"ssh_key": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"passphrase": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"store_in_statefile": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		}),
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+			_, password_ok := diff.GetOk("password")
+			_, ssh_key_ok := diff.GetOk("ssh_key")
+			if password_ok && ssh_key_ok {
+				return errors.New(`"password" and "ssh_key" cannot be provided in the same resource.`)
+			}
+			if !password_ok && !ssh_key_ok {
+				return errors.New(`"password" or "ssh_key" must be provided in the same resource.`)
+			}
+			return nil
+		},
+	}
+}
+
+func setAuthenticationInState(d *schema.ResourceData) {
+
+	d.Set("password", "")
+	d.Set("ssh_key", "")
+	d.Set("passphrase", "")
```

Comment:
This may cause an issue when store_in_statefile is true.

Reply:
you are correct should be in conditional, changed it